Jolt/Core/Core.h

```diff
@@ -4,6 +4,22 @@
 
 #pragma once
 
+// If this define is set, Jolt is compiled as a shared library
+#ifdef JPH_SHARED_LIBRARY
+	#ifdef JPH_BUILD_SHARED_LIBRARY
+		// While building the shared library, we must export these symbols
+		#define JPH_EXPORT __declspec(dllexport)
```

Comment:
How compatible is this with Linux? Does everything need to change if someone asks for a shared library on Linux?

Reply:
Looking at our code, in Linux you would use "[[gnu::visibility("default")]]" rather than __declspec(dllimport)/__declspec(dllexport). The rest stays the same.

Reply:
Don't take my word for this, but I believe you should also (ideally) pair that up with the compiler flag `-fvisibility=hidden` on Linux, to stop it from bundling any symbols that aren't explicitly exported. See [this link](https://gcc.gnu.org/wiki/Visibility) for more details, as well examples of cross-platform export macros.

Reply:
I tried ```[[gnu::visibility("default")]]``` yesterday and then clang started generating errors that this attribute is invalid on a class but today it works without errors. @mihe I followed the steps you list and now it produces a .so on Linux, thanks!